src/databricks/labs/ucx/account.py

```diff
@@ -38,29 +38,33 @@ def _get_cloud(self) -> str:
     def client_for(self, workspace: Workspace) -> WorkspaceClient:
         return self._ac.get_workspace_client(workspace)
 
-    def workspace_clients(self) -> list[WorkspaceClient]:
+    def workspace_clients(self, workspaces: list[Workspace]) -> list[WorkspaceClient]:
         """
         Return a list of WorkspaceClient for each configured workspace in the account
         :return: list[WorkspaceClient]
         """
         clients = []
-        for workspace in self._workspaces():
+        for workspace in workspaces:
             ws = self.client_for(workspace)
             clients.append(ws)
         return clients
 
-    def sync_workspace_info(self):
+    def sync_workspace_info(self, workspaces: list[Workspace] | None = None):
```

Comment:
Does the cli command still work? I'm surprised that no test failed on this change to public api

Reply:
the new param is optional, so does not affect existing behaviour